src/transport/websocket/mod.rs

```diff
@@ -618,22 +603,15 @@ impl Stream for WebSocketTransport {
                         endpoint,
                     }));
                 }
-                Err(error) => match error.connection_id {
-                    Some(connection_id) => match self.pending_dials.remove(&connection_id) {
-                        Some(address) =>
-                            return Poll::Ready(Some(TransportEvent::DialFailure {
-                                connection_id,
-                                address,
-                                error: error.error,
-                            })),
-                        None => {
-                            tracing::debug!(target: LOG_TARGET, ?error, "failed to establish connection")
-                        }
-                    },
-                    None => {
-                        tracing::debug!(target: LOG_TARGET, ?error, "failed to establish connection")
```

Comment:
nit: May be keep these log messages to catch the state mismatch if it happens? May be even add `debug_assert!(false)`?

Reply:
After looking a bit into this, we only have a `pending_dial` associated with the connection ID if the connection was outbound. For inbound connections, we don't track which address failed here